jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/UserConfigurationCompareTest.java

```diff
@@ -0,0 +1,158 @@
+        //Then
+        assertTrue(result);
+    }
+    @Test
+    public void TestConfigEqualWhenMapAndNotEqual() {
+        //Given
+        Map<String, String> entity1 = new HashMap();
+        Map<String, String> entity2 = new HashMap();
+        entity1.put("item", "code");
+        entity2.put("item1", "code");
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1, entity2);
+        //Then
+        assertFalse(result);
+    }
+
+
+    @Test
+    public void TestConfigEqualWhenObjectMetaIsTrue() {
+        //Given
+        Object entity1 = new ObjectMetaBuilder().withName("test1").withAnnotations(Collections.singletonMap("foo", "bar")).build();
+        Object entity2 = new ObjectMetaBuilder().withName("test1").withAnnotations(Collections.singletonMap("foo","bar")).build();
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1, entity2);
+        //Then
+        assertTrue(result);
+    }
+
+    @Test
+    public void TestConfigEqualWhenCollectionWhenFalse(){
+        //Given
+        ArrayList<String> anandList = new ArrayList<>();
+        Object entity1 = Collections.singletonList(anandList);
+        Object entity2 = Collections.EMPTY_LIST;
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1,entity2);
+        //Then
+        assertFalse(result);
+    }
+
+    @Test
+    public void TestConfigEqualWhenCollectionWhenTrue(){
+        //Given
+        Object entity1 = Collections.EMPTY_LIST;
+        Object entity2 = Collections.EMPTY_SET;
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1,entity2);
+        //Then
+        assertTrue(result);
+    }
+
+
+    @Test
+    public void TestConfigEqual() {
+        //Given
+        Object entity1 = "asda";
+        Object entity2 = "asdasda";
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1, entity2);
+        //Then
+        assertFalse(result);
+    }
+
+    @Test
+    public void TestConfigEqualWhenKDTO() {
+        //Given
+        Object entity1 = new UserConfigurationCompare();
+        Object entity2 = new UserConfigurationCompare();
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1, entity2);
+        //Then
+        assertTrue(result);
+    }
+    @Test
+    public void TestConfigEqualWhenNotKDTO() {
+        //Given
+        Object entity1 = new KubernetesListBuilder().addToItems(new DeploymentBuilder().build()); //
+        Object entity2 = new KubernetesListBuilder().addToItems(new DeploymentBuilder().build()); //
+        //When
+        boolean result = UserConfigurationCompare.configEqual(entity1, entity2);
+        //Then
+        assertTrue(result);
+    }
+    @Test
+    public void TestConfigEqualWhenKDTOWithObj() {
+        //Given
+        Object entity1 = new UserConfigurationCompare();
```

Comment:
In my opinion it's not a good idea do create an object of `UserConfigurationCompare` . You should use some actual Kubernetes Objects using builders like this:
```
    Deployment deployment1 = new DeploymentBuilder().withNewMetadata()
      .withName("deployment1")
      .addToLabels("key1", "value1")
      .withResourceVersion("1")
      .withGeneration(1L)
      .endMetadata()
      .withNewSpec()
      .withNewSelector()
      .addToMatchLabels("key1", "value1")
      .endSelector()
      .withReplicas(0)
      .endSpec()
      .withNewStatus()
      .withReplicas(1)
      .withObservedGeneration(1L)
      .endStatus()
      .build();
    Deployment deployment1 = new DeploymentBuilder().withNewMetadata()
      .withName("deployment1")
      .addToLabels(Collections.singletonMap("key1", "value1"))
      .withResourceVersion("1")
      .withGeneration(1L)
      .endMetadata()
      .withNewSpec()
      .withNewSelector()
      .addToMatchLabels("key1", "value1")
      .endSelector()
      .withReplicas(0)
      .endSpec()
      .withNewStatus()
      .withReplicas(1)
      .withObservedGeneration(1L)
      .endStatus()
      .build();
     // When
     boolean result = UserConfigurationCompare.configEqual(deployment1, deployment2);
     // Then
     assertFalse(result);
```